components/script/dom/htmlimageelement.rs

```diff
@@ -832,7 +849,14 @@ impl HTMLImageElement {
                         self.image_request.set(ImageRequestPhase::Pending);
                         self.init_image_request(&mut pending_request, url, src);
                     },
-                    (_, State::Broken) | (_, State::Unavailable) => {
+                    (_, State::Broken) => {
+                        self.reject_image_decode_promises();
```

Comment:
I think this one also does not belong here, because the request is already in the broken state.